app/src/main/java/zapsolutions/zap/pin/PinSetupActivity.java

```diff
@@ -0,0 +1,133 @@
+package zapsolutions.zap.pin;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import zapsolutions.zap.HomeActivity;
+import zapsolutions.zap.R;
+import zapsolutions.zap.baseClasses.App;
+import zapsolutions.zap.baseClasses.BaseAppCompatActivity;
+import zapsolutions.zap.connection.manageWalletConfigs.Cryptography;
+import zapsolutions.zap.util.PrefsUtil;
+import zapsolutions.zap.util.TimeOutUtil;
+import zapsolutions.zap.util.UtilFunctions;
+
+
+public class PinSetupActivity extends BaseAppCompatActivity implements PinActivityInterface {
+
+    public static final int ADD_PIN = 0;
+    public static final int CHANGE_PIN = 1;
+
+    private static final String LOG_TAG = PinSetupActivity.class.getName();
+
+    private Fragment mCurrentFragment = null;
+    private FragmentTransaction mFt;
+    private int mSetupMode = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_setup);
+
+        // Receive data from last activity
+        Bundle extras = getIntent().getExtras();
+        if (extras != null) {
+            mSetupMode = extras.getInt("setupMode", 0);
+        }
+
+
+        // Set pin fragment as beginning fragment
+        showCreatePin();
+
+        switch (mSetupMode) {
+            case ADD_PIN:
+                showCreatePin();
+                break;
+            case CHANGE_PIN:
+                showEnterPin();
+                break;
+        }
+    }
+
+    public void pinCreated(String value) {
+        App.getAppContext().pinTemp = value;
```

Comment:
I think this would be a good time to remove those members from the `App` class and forward the pin with bundle extras to the fragment. What do you think?